src/ng/animate.js

```diff
@@ -416,8 +416,8 @@ var $AnimateProvider = ['$provide', /** @this */ function($provide) {
        * @param {Promise} animationPromise The animation promise that is returned when an animation is started.
        */
       cancel: function(runner) {
-        if (runner.end) {
-          runner.end();
+        if (runner.cancel) {
+          runner.cancel();
```

Comment:
If $animate.cancel doesn't reject the promise, then .done callback handlers aren't called with false, which means in the ngSwitch done handler, I cannot distinguish between failed and successful animations, which leads to "cannot read function .end of undefined" error I had at the beginning. I then changed the code in ngSwitch to read the stored runners from the end to work around this problem, but then you said, I should check for successful animations inside done, which is how I found this problem with cancel in the first place